### DISCUSSION TITLE
What about multiple users on the same device?

### DISCUSSION THREAD

**Discussion by Eraldo** [xtrinch/fcm-django#264] [Q&A]
I would like to support multiple users on the same physical device.
Since tokens are user agnostic, that means I will get the same token.

There are two possible paths that I see:
A) Change `user` to `users` on the Device model. (I.e. Many-to-Many)
B) Remove the unique constraint of `registration_id` and instead use `unique_together = ("registration_id", "user")`

What are your thoughts?

**Comment by xtrinch** [comment]
Since to be able to use the same app for multiple users you need to log out in between - and on log out you should clean up the tokens (delete the device). I don't see how you would run into ever needing multiple logged in users at the same time - do you have a more specific use case why you would want this?

**Reply by Eraldo** [reply]
Just like youtube itself, I allow more than one user to be logged in at the same time without having to log out the previous user.
(In my specific example I personally have a private and a professional user logged in at the same time for example.)

**Reply by xtrinch** [reply]
In this case, I think B) option would be viable -> B) Remove the unique constraint of registration_id and instead use unique_together = ("registration_id", "user")

**Reply by Eraldo** [reply]
I agree. This use case might also be considered for the swappable model approach.
Ref: https://github.com/xtrinch/fcm-django/pull/239
